megamek/src/megamek/common/verifier/TestSmallCraft.java

```diff
@@ -374,7 +374,17 @@ public boolean isAero() {
     public boolean isSmallCraft() {
         return true;
     }
-
+    
+    //Determines if we have a Naval C3 installed. Needed for FCS tonnage calculations.
+    public static boolean hasNC3(SmallCraft sc) {
+        for (Mounted m : sc.getEquipment()) {
```

Comment:
Rather than iterating through weapons and ammo along with MiscType, you can just look through the MiscType list with `Entity.getMisc()`. You can also replace this whole construction with `sc.hasWorkingMisc(MiscType.F_NAVAL_C3)`. It adds the extra step of checking whether the equipment is operational, but this is not a place where the very slight performance hit will matter and validation is only done on newly loaded units. There are other places in the validation code where the same assumption is made.